src/v/bytes/iobuf.h

```diff
@@ -425,9 +426,9 @@ inline void iobuf::append(iobuf&& o) {
             }
         }
 
-        append(std::move(f).unoptimized_release());
+        append(f);
     }
-    o.clear();
+    o._size = 0;
```

Comment:
This drops the the `update_generation()` call which we shouldn't do (I guess there is no meta test for this since one didn't fail here).

Is this noticeably faster than `clear()`? If so maybe we just add the `update_generation()` call back in explicitly.

Reply:
Let me check if it's any faster.

Reply:
The effects appear to be negligible, will switch back to using `iobuf::clear()`

Reply:
Yeah this seems more robust to future changes also.